src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs

```diff
@@ -65,7 +76,7 @@ private static void HandleBinaryPattern(IBinaryPatternOperation? binaryPattern,
 
         private static void RemoveIfConstantPatternHasValue(IOperation operation, Dictionary<long, ISymbol> enumMembers)
         {
-            if (operation is IConstantPatternOperation { Value.ConstantValue: { HasValue: true, Value: var value } })
+            if (operation is IConstantPatternOperation { Value.ConstantValue: { HasValue: true, Value: var value } } && value is not null)
```

Comment:
nit:
```suggestion
            if (operation is IConstantPatternOperation { Value.ConstantValue: { HasValue: true, Value: not null and var value } })
```

up to you :)

Reply:
Uh, this fancy patterns)...